### PR TITLE
Allow invitation-only enrollments

### PR DESCRIPTION
**Pull request by Ivanca**
We had to copy the internals of the create_course_enrollment function to use check_access=False instead of check_access=True when calling CourseEnrollment.enroll

### CODE REVIEW

**Comment by Ivanca** [comment]
@felipemontoya Feedback changes done at aebb7aaf9af960b3880be04ca6da56e62553fd19

File: eox_core/edxapp_wrapper/backends/enrollment_h_v1.py [hunk 1]
```diff
@@ -28,24 +26,40 @@ def create_enrollment(*args, **kwargs):
...
+                user = User.objects.get(username=username)
+                enrollment = CourseEnrollment.enroll(user, course_key, check_access=False)
+                api._data_api()._update_enrollment(enrollment, is_active=is_active, mode=mode)
+            except Exception as e:
+                raise APIException(repr(e))
+        else:
+            raise APIException(repr(e))
+
+    if enrollment_attributes is not None:
+        api.set_enrollment_attributes(username, course_id, enrollment_attributes)
```
Comment (felipemontoya) [inline eox_core/edxapp_wrapper/backends/enrollment_h_v1.py]: Any chance that this is the line that fails?

Comment (Ivanca) [reply eox_core/edxapp_wrapper/backends/enrollment_h_v1.py]: not really

File: eox_core/edxapp_wrapper/backends/enrollment_h_v1.py [hunk 2]
```diff
@@ -67,3 +70,25 @@ def check_edxapp_enrollment_is_valid(*args, **kwargs):
...
+        if try_update:
+            enrollment = api._data_api().update_course_enrollment(username, course_id, mode, is_active)
+        else:
+            raise Exception(repr(e) + ", use force_registration to update the existing enrollment")
+    return enrollment
+
+
+def _force_create_enrollment(username, course_id, mode, is_active):
+    try:
+        course_key = CourseKey.from_string(course_id)
```
Comment (felipemontoya) [inline eox_core/edxapp_wrapper/backends/enrollment_h_v1.py]: Could you please log here under the INFO level that we used the force enrollment creation?

File: eox_core/edxapp_wrapper/backends/enrollment_h_v1.py [hunk 3]
```diff
@@ -67,3 +70,25 @@ def check_edxapp_enrollment_is_valid(*args, **kwargs):
...
+        user = User.objects.get(username=username)
+        enrollment = CourseEnrollment.enroll(user, course_key, check_access=False)
+        api._data_api()._update_enrollment(enrollment, is_active=is_active, mode=mode)
+    except Exception as e:
+        raise APIException(repr(e))
```
Comment (felipemontoya) [inline eox_core/edxapp_wrapper/backends/enrollment_h_v1.py]: Please log here if something failed miserably with WARN or ERROR

Comment (Ivanca) [reply eox_core/edxapp_wrapper/backends/enrollment_h_v1.py]: @felipemontoya  done

**Pull request merged**